include/http_base.h

```diff
@@ -10,6 +10,14 @@
 
 namespace azure {  namespace storage_lite {
 
+    // Case insensitive comparator
+    struct CaseInsensitiveCompare 
```

Comment:
I believe this is a more universal utility that can be moved to `utility.h`. Also, to align with the name convention in this repository, can you please rename it into `case_insensitive_compare`?

Reply:
Thanks and agreed on both of those. Had an issue with order and circular dependency as `http_base.h` is included in `utility.h`. I moved to `compare.h` to resolve and still seems valid for it being a utility under it's own domain?